src/api_manager/fetch_metadata.cc

```diff
@@ -132,32 +134,35 @@ void GlobalFetchServiceAccountToken(
       env->LogDebug("Need to fetch service account token");
   }
 
+  std::shared_ptr<cloud_trace::CloudTraceSpan> trace_span(
+      CreateSpan(cloud_trace, trace_name));
   token->set_state(auth::ServiceAccountToken::FETCHING);
-  FetchMetadata(context.get(), path, kMetadataTokenFetchRetries,
-                [env, token, continuation, audience](
-                    Status status, std::map<std::string, std::string> &&,
-                    std::string &&body) {
-                  // fetch failed
-                  if (!status.ok()) {
-                    env->LogError("Failed to fetch service account token");
-                    token->set_last_failed_fetch_time(system_clock::now());
-                    token->set_state(auth::ServiceAccountToken::FAILED);
-                    continuation(Status(Code::INTERNAL, kFailedTokenFetch));
-                    return;
-                  }
-                  if (audience.empty()) {
-                    if (!token->SetTokenJsonResponse(body)) {
-                      env->LogError("Failed to parse token response body");
-                      continuation(Status(Code::INTERNAL, kFailedTokenParse));
-                      return;
-                    }
-                  } else {
-                    // TODO: parse JWT to get expiration time.
-                    token->set_access_token(body,
-                                            kInstanceIdentityTokenExpiration);
-                  }
-                  continuation(Status::OK);
-                });
+  FetchMetadata(
```

Comment:
Does `FetchMetadata` do retries? I assume we won't create multiple traces if that occurs. Maybe that is OK, just curious.

Reply:
It will do retry.  But all retries will be in the same span,